neuro_data/static_images/configs.py

```diff
@@ -217,4 +217,61 @@ def load_data(self, key, tier=None, batch_size=1, key_order=None, stimulus_types
                 dataset.transforms.insert(-1, Subsample(np.where(selection)[0]))
 
                 assert np.all(dataset.neurons.unit_ids == units[selection]), 'Units are inconsistent'
-            return datasets, loaders
\ No newline at end of file
+            return datasets, loaders
+
+    class AreaLayerNoise(dj.Part, AreaLayerRawMixin):
+        definition = """
+        -> master
+        ---
+        stats_source            : varchar(50)   # normalization source
+        stimulus_type           : varchar(50)   # type of stimulus
+        exclude                 : varchar(512)  # what inputs to exclude from normalization
+        normalize               : bool          # whether to use a normalize or not
+        -> experiment.Layer
+        -> anatomy.Area
+        """
+
+        def describe(self, key):
+            return "{brain_area} {layer} on {stimulus_type}. normalize={normalize} on {stats_source} (except '{exclude}')".format(**key)
+
+        @property
+        def content(self):
+            for p in product(['all'],
+                             ['~stimulus.Frame'],
+                             ['images,responses',''],
+                             [True],
+                             ['L4', 'L2/3'],
+                             ['V1', 'LM']):
+                yield dict(zip(self.heading.dependent_attributes, p))
+
+        def load_data(self, key, test_index=0, **kwargs):
+            if 'tier' in kwargs.keys():
+                tier = kwargs['tier']
+                kwargs['tier'] = None
+            else:
+                tier = None
+            datasets, loaders = super().load_data(key, **kwargs)
+            if tier is not None:
+                log.info('Filtering by tier={}'.format(tier))
+                for k, dataset in datasets.items():
+                    noise_indices = np.flatnonzero(dataset.types != 'stimulus.Frame')
+                    unique_condition_hashes = np.unique(dataset.info.condition_hash[noise_indices])
+                    assert test_index < unique_condition_hashes.size
+                    if tier == 'test':
+                        tier_condition_hashes = np.array([unique_condition_hashes[test_index]])
+                    else:
+                        train_val_indices = np.flatnonzero(np.arange(unique_condition_hashes.size) != test_index)
+                        train_size = np.round(train_val_indices.size * .95).astype(np.int)
+                        np.random.seed(test_index)
```

Comment:
I'd usually want to use a configurable seed. Perhaps consider having `seed` as a field for this table? Same thing goes with the `train`/`validation` fraction.